packages/file-explorer/src/FileExplorer.test.tsx

```diff
@@ -0,0 +1,193 @@
+import React from 'react';
+import {
+  render,
+  screen,
+  waitForElementToBeRemoved,
+} from '@testing-library/react';
+import userEvent from '@testing-library/user-event/';
+import { MockFileStorage } from '@deephaven/dashboard-core-plugins';
+import { ContextMenuRoot } from '@deephaven/components';
+import { TestUtils } from '@deephaven/utils';
+import FileStorage, {
+  DirectoryStorageItem,
+  FileStorageItem,
+} from './FileStorage';
+import FileExplorer, { FileExplorerProps } from './FileExplorer';
+import FileTestUtils from './FileTestUtils';
+
+function makeMockFileStorage(): FileStorage {
+  return {
+    getTable: jest.fn(),
+    saveFile: jest.fn(),
+    loadFile: jest.fn(),
+    deleteFile: jest.fn(),
+    moveFile: jest.fn(),
+    info: jest.fn(),
+    createDirectory: jest.fn(),
+  };
+}
+
+function renderFileExplorer({
+  storage,
+  onSelect = jest.fn(),
+  onRename = jest.fn(),
+  onDelete = jest.fn(),
+}: Partial<FileExplorerProps>) {
+  if (storage !== undefined) {
+    return render(
+      <>
+        <FileExplorer
+          storage={storage}
+          onSelect={onSelect}
+          onRename={onRename}
+          onDelete={onDelete}
+        />
+        <ContextMenuRoot />
+      </>
+    );
+  }
+}
+
+it('mounts and unmounts successfully without crashing', () => {
+  renderFileExplorer({ storage: makeMockFileStorage() });
+});
+
+it('mounts properly and shows file list', async () => {
+  const files = FileTestUtils.makeFiles();
+
+  const fileStorage = new MockFileStorage(files);
+
+  renderFileExplorer({ storage: fileStorage as FileStorage });
+
+  const foundItems = await screen.findAllByRole('listitem');
+  expect(foundItems).toHaveLength(files.length);
+});
+
+describe('context menu actions work properly', () => {
+  let user: ReturnType<typeof userEvent.setup>;
+  let dirs: DirectoryStorageItem[] = [];
+  let files: FileStorageItem[] = [];
+  let items: FileStorageItem[] = [];
+  const onRename = jest.fn();
+  const onDelete = jest.fn();
+
+  const clickContextMenuButton = async (
+    itemName: RegExp,
+    buttonName: RegExp
```

Comment:
You could have these both accept `RegExp | string` so you could just pass a string in here instead of always having to pass in regexes.